File: tests/human/x/erlang/cross_join_filter.erl
#!/usr/bin/env escript
%% cross_join_filter.erl - manual translation of tests/vm/valid/cross_join_filter.mochi

main(_) ->
    Nums = [1,2,3],
    Letters = ["A","B"],
    Pairs = [#{n => N, l => L} || N <- Nums, L <- Letters, N rem 2 =:= 0],
    io:format("--- Even pairs ---~n"),
    lists:foreach(fun(P) ->
        io:format("~p ~s~n", [maps:get(n, P), maps:get(l, P)])
    end, Pairs).

File: tests/human/x/dart/cross_join_filter.dart
void main() {
  var nums = [1, 2, 3];
  var letters = ['A', 'B'];
  var pairs = <Map<String, dynamic>>[];
  for (var n in nums) {
    if (n % 2 == 0) {
      for (var l in letters) {
        pairs.add({'n': n, 'l': l});
      }
    }
  }
  print('--- Even pairs ---');
  for (var p in pairs) {
    print('${p['n']} ${p['l']}');
  }
}
